mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -189,11 +227,20 @@ class ChiSqSelector @Since("1.3.0") (
    */
   @Since("1.3.0")
   def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
-    val indices = Statistics.chiSqTest(data)
-      .zipWithIndex.sortBy { case (res, _) => -res.statistic }
-      .take(numTopFeatures)
-      .map { case (_, indices) => indices }
-      .sorted
+    var indices = selectorType match {
+      case ChiSqSelectorType.KBest => Statistics.chiSqTest(data)
+        .zipWithIndex.sortBy { case (res, _) => -res.statistic }
```

Comment:
I think we want to compute and sort the chi squared stats in all cases, not just two. This can be lifted out.